File: kb/parser/parser_samples/sample.jl
module Sample

struct Point
    x::Int
    y::Int
end

# adds two points
function add(p1::Point, p2::Point)::Point
    return Point(p1.x + p2.x, p1.y + p2.y)
end

p = add(Point(1, 2), Point(3, 4))
println(p)

end # module

File: kb/parser/parser_samples/sample.py
from __future__ import annotations
from dataclasses import dataclass


@dataclass
class Point:
    x: int
    y: int

    def add(self, other: "Point") -> "Point":
        return Point(self.x + other.x, self.y + other.y)


def distance(p1: Point, p2: Point) -> float:
    """Euclidean distance between two points."""
    return ((p1.x - p2.x) ** 2 + (p1.y - p2.y) ** 2) ** 0.5


if __name__ == "__main__":
    print(distance(Point(0, 0), Point(3, 4)))
